src/rebar3-nix-bootstrap

```
@@ -0,0 +1,219 @@
+%% would be `faz/foo/foos/bash`.
+-spec make_symlink(string(), string()) -> ok.
+make_symlink(Path, Target) ->
+    AppName = filename:basename(Path),
+    TargetFile = filename:join(Target, fixup_app_name(AppName)),
+    filelib:ensure_dir(TargetFile),
+    ok = case filelib:is_file(TargetFile) of
+             true ->
+                 file:delete(TargetFile);
+             false ->
+                 ok
+         end,
+    io:format("Making symlink from ~s to ~s~n", [Path, TargetFile]),
+    file:make_symlink(Path, TargetFile).
+
+%% @doc
+%% This takes an app name in the standard OTP <name>-<version> format
+%% and returns just the app name. Why? because rebar is doesn't
+%% respect OTP conventions in some cases.
+-spec fixup_app_name(string()) -> string().
+fixup_app_name(FileName) ->
+    [Name, _Version] = string:tokens(FileName, "-"),
+    Name.
+
+-spec bootstrap_registry(#data{}) -> ok.
+bootstrap_registry(#data{registry_snapshot = RegistrySnapshot}) ->
+    io:format("Bootstrapping Hex Registry for Rebar~n"),
+    make_sure_registry_snapshot_exists(RegistrySnapshot),
+    filelib:ensure_dir(?HEX_REGISTRY_PATH),
+    ok = case filelib:is_file(?HEX_REGISTRY_PATH) of
+             true ->
+                 file:delete(?HEX_REGISTRY_PATH);
+             false ->
+                 ok
+         end,
+    ok = file:make_symlink(RegistrySnapshot,
+                           ?HEX_REGISTRY_PATH).
+
+-spec make_sure_registry_snapshot_exists(string()) -> ok.
+make_sure_registry_snapshot_exists(RegistrySnapshot) ->
+    case filelib:is_file(RegistrySnapshot) of
+        true ->
+            ok;
+        false ->
+            stderr("Registry snapshot (~s) does not exist!", [RegistrySnapshot])
+    end.
+
+-spec gather_required_data_from_the_environment(#data{}) -> {ok, map()}.
+gather_required_data_from_the_environment(ArgData) ->
+    {ok, ArgData#data{ version = guard_env("version")
+                     , erl_libs = os:getenv("ERL_LIBS", [])
```

Based on https://github.com/NixOS/nixpkgs/compare/master...ericbmerritt:erlang-improvements#diff-85631298b9cdea5c4f43bb9be3fd5a93R37 it seem that ERL_LIBS should always be defined if the tool is called correctly, even if there are no deps. I prefer stricter code when possible, since it'll fail faster, but it's up to you.